frb/dm/prob_dmz.py

```diff
@@ -169,4 +173,67 @@ def grab_repo_grid():
     sdict = np.load(PDM_z_grid_file)
 
     # Return
-    return sdict
\ No newline at end of file
+    return sdict
+
+
+def get_DMcosmic_from_z(zarray, perc=0.5, DMevals=np.linspace(1.,2000.,1000), beta=3., F=0.31, cosmo=defs.frb_cosmo):
+    """
+    Gives DMcosmic values of zarray, considering the percentile.
+    Default is median (i.e. percentile=0.5)
+
+    Parameters
+    ----------
+    zarray : float or np.array
+        Redshift value or values
+        If np.array, then it must start from 0
+    perc : float
+        Percentile of the PDF(DM_cosmic) for each z, from 0 to zmax. 
+        Default = 0.5
+    DMevals : np.array
+        Array representing the DMcosmic evaluations of the PDF. Should start with 1.
+        Default = np.linspace(1.,5000.,1000)
+    beta : float
+        Slope of a galactic halo parameter (See Macquart+2020)
+        Default = 3.0
+    F : float
+        Parameter that depends on galaxy feedback (See Macquart+2020)
+        Default = 0.31
+    cosmo : astropy.cosmology object
+        Cosmology
+
+
+    Returns
+    -------
+    zarray : np.array
+        z values where the DMcosmic was computed
+    DMcosmic_array : np.array
+        DMcosmic values corresponding to the z_array (in a given percentile)
+    
+    """
+    
+    #check z input
+    if isinstance(zarray, float):
+        z_new = np.array([0,zarray])
+    else:
+        z_new = np.array(zarray)
+   
+    # Get the DMcosmic-z grid
+    # This is time consuming, could be done more efficiently
+    print("Calculating the DMcosmic-z grid, this may take a while...")
+    zeval , DM_cosmics, PDF_grid = grid_P_DMcosmic_z(zvals=z_new, beta=beta, F=F, cosmo=cosmo, DM_cosmics=DMevals)
```

Comment:
Now that I think about it, maybe it would be a good idea to just store the default grid of values in a compressed `npz` file. For a 1000x1000 grid of floats, the size should be less than 1 MB. Users wanting to generate custom grids can do so at their leisure.

Alternatively, maybe provide an option to read the grid from a file.